### PR TITLE
Implementation UTC_DATE/UTC_TIME/UTC_TIMESTAMP

### PR DESCRIPTION
**Pull request by MitchellGale-BitQuill**
### Description
UTC_DATE, UTC_TIME, UTC_TIMESTAMP are all implemented according to MySQL standard.
 
### Issues Resolved
[Support date and time functions for new engine · Issue #709 · opendistro-for-elasticsearch/sql](https://github.com/opendistro-for-elasticsearch/sql/issues/709) 

[Date time query improvement · Issue #46 · opensearch-project/sql](https://github.com/opensearch-project/sql/issues/46)
 
```
opensearchsql> select utc_date();                                                                                                                                                  
fetched rows / total rows = 1/1
+--------------+
| utc_date()   |
|--------------|
| 2022-10-03   |
+--------------+
opensearchsql> select utc_time();                                                                                                                                                  
fetched rows / total rows = 1/1
+--------------+
| utc_time()   |
|--------------|
| 17:54:27     |
+--------------+
opensearchsql> select utc_timestamp();                                                                                                                                             
fetched rows / total rows = 1/1
+---------------------+
| utc_timestamp()     |
|---------------------|
| 2022-10-03 17:54:28 |
+---------------------+
```

### Check List
- [X] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 1]
```diff
@@ -974,6 +1002,38 @@ private ExprValue exprTimeToSec(ExprValue time) {
     return new ExprLongValue(time.timeValue().toSecondOfDay());
   }
 
+  /**
+   * To_days implementation for ExprValue.
+   *
+   * @return ExprValue.
```
Comment (raymond-lum) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1008]: JavaDoc should explain what the return is.

**Review by raymond-lum — COMMENTED** [review]
Please add unit tests.

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+
+@ExtendWith(MockitoExtension.class)
+class UTCFunctionsTest extends ExpressionTestBase {
+  int allowedSecondsOffset = 60;
+
+  @Mock
+  Environment<Expression, ExprValue> env;
+
+  @Test
```
Comment (raymond-lum) [inline core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java:38]: What is this supposed to test?

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    LocalDateTime result = expr.valueOf(env).datetimeValue();
+    assertEquals(TIMESTAMP, expr.type());
+
+    LocalDateTime currentUTC = LocalDateTime.now(ZoneOffset.UTC);
+    LocalDateTime lowerUTC = currentUTC.minus(allowedSecondsOffset, ChronoUnit.SECONDS);
+    LocalDateTime upperUTC = currentUTC.minus(-allowedSecondsOffset, ChronoUnit.SECONDS);
+
+    if (!lowerUTC.isBefore(result) || !upperUTC.isAfter(result)) {
+      assertTrue(Boolean.parseBoolean("TimeStamp was not within range"));
+    }
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java:50]: You can simplify the test by using
```
assertEquals(SECONDS.between(currentUTC, result), 0, allowedSecondsOffset);
```
The third argument for `assertEquals` is delta.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 4]
```diff
@@ -974,6 +1002,41 @@ private ExprValue exprTimeToSec(ExprValue time) {
     return new ExprLongValue(time.timeValue().toSecondOfDay());
   }
 
+  /**
+   * UTC_DATE implementation for ExprValue.
+   * Returns in format `yyyy-MM-dd`.
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1007]: Doesn't this return a ExprDateValue?  Your comment suggests that it's a string with the format `yyyy-mm-dd`.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 5]
```diff
@@ -974,6 +1002,41 @@ private ExprValue exprTimeToSec(ExprValue time) {
...
+   * @return ExprValue.
+   */
+  private ExprValue exprUtcDate() {
+    return new ExprDateValue(exprUtcTimeStamp().dateValue());
+
+  }
+
+  /**
+   * UTC_TIME implementation for ExprValue.
+   * Returns in format `HH:mm:ss`.
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1018]: Same here.  I think this returns a ExprTimeValue, not a string with a format.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 6]
```diff
@@ -974,6 +1002,41 @@ private ExprValue exprTimeToSec(ExprValue time) {
...
+   *
+   * @return ExprValue.
+   */
+  private ExprValue exprUtcTime() {
+    return new ExprTimeValue(exprUtcTimeStamp().timeValue());
+  }
+
+  /**
+   * UTC_TIMESTAMP implementation for ExprValue.
+   * Returns in format `yyyy-MM-dd HH:mm:ss`.
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1028]: Same here - this returns a exprDateTime, not a string with format

File: docs/user/dql/functions.rst [hunk 7]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
 
 
+
+UTC_DATE
+-------
+
+Description
+>>>>>>>>>>>
+
+Returns the current UTC date as a value in 'yyyy-MM-dd'.
```
Comment (acarbonetto) [inline docs/user/dql/functions.rst:1721]: Documentation above says: `YYYY-MM-DD`
Is it better to be consistent?

File: docs/user/dql/functions.rst [hunk 8]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+
+
+
+UTC_TIMESTAMP
+-------
+
+Description
+>>>>>>>>>>>
+
+Returns the current UTC timestamp as a value in 'yyyy-MM-dd HH:mm:ss'.
```
Comment (acarbonetto) [inline docs/user/dql/functions.rst:1770]: Documentation above says: YYYY-MM-DD
Is it better to be consistent?

File: docs/user/ppl/functions/datetime.rst [hunk 9]
```diff
@@ -1152,6 +1152,81 @@ Example::
     +--------------------------+-----------------------------+
 
 
+UTC_DATE
+-------
+
+Description
+>>>>>>>>>>>
+
+Returns the current UTC date as a value in 'yyyy-MM-dd'.
```
Comment (acarbonetto) [inline docs/user/ppl/functions/datetime.rst:1161]: Documentation above says: YYYY-MM-DD
Is it better to be consistent?

File: docs/user/ppl/functions/datetime.rst [hunk 10]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+
+
+
+UTC_TIMESTAMP
+-------
+
+Description
+>>>>>>>>>>>
+
+Returns the current UTC timestamp as a value in 'yyyy-MM-dd HH:mm:ss'.
```
Comment (acarbonetto) [inline docs/user/ppl/functions/datetime.rst:1210]: Documentation above says: YYYY-MM-DD
Is it better to be consistent?

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 11]
```diff
@@ -974,6 +1002,41 @@ private ExprValue exprTimeToSec(ExprValue time) {
...
+   *
+   * @return ExprValue.
+   */
+  private ExprValue exprUtcTimeStamp() {
+    String formattedLDT = LocalDateTime.now().format(DATE_TIME_FORMATTER_NON_STRUCT);
+    ExprStringValue localDateValue = new ExprStringValue(formattedLDT);
+    ExprStringValue toTZ = new ExprStringValue("+00:00");
+
+    return exprDateTime(localDateValue, toTZ);
```
Comment (acarbonetto) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1037]: Is this supposed to return a Datetime or Timestamp?

File: docs/user/dql/functions.rst [hunk 12]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+
+Return type: DATETIME
+
+Specification: UTC_TIMESTAMP() -> DATETIME
```
Comment (acarbonetto) [inline docs/user/dql/functions.rst:1774]: Feels like this should be returning a `TIMESTAMP`?

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 [hunk 13]
```diff
@@ -398,7 +398,7 @@ dateAndTimeFunctionBase
     : ADDDATE | CONVERT_TZ | DATE | DATETIME | DATE_ADD | DATE_FORMAT | DATE_SUB | DAY | DAYNAME | DAYOFMONTH | DAYOFWEEK
     | DAYOFYEAR | FROM_DAYS | FROM_UNIXTIME | HOUR | MAKEDATE | MAKETIME | MICROSECOND | MINUTE
     | MONTH | MONTHNAME | QUARTER | SECOND | SUBDATE | SYSDATE | TIME | TIMESTAMP | TIME_TO_SEC
-    | TO_DAYS | UNIX_TIMESTAMP | WEEK | YEAR
+    | TO_DAYS | UNIX_TIMESTAMP | UTC_DATE | UTC_TIME | UTC_DATETIME | WEEK | YEAR
```
Comment (acarbonetto) [inline ppl/src/main/antlr/OpenSearchPPLParser.g4:401]: why is it called UTC_DATETIME (vs UTC_TIMESTAMP) for the PPL parser?

File: sql/src/main/antlr/OpenSearchSQLParser.g4 [hunk 14]
```diff
@@ -397,7 +397,7 @@ dateTimeFunctionName
     : ADDDATE | CONVERT_TZ | DATE | DATETIME | DATE_ADD | DATE_FORMAT | DATE_SUB | DAY | DAYNAME | DAYOFMONTH | DAYOFWEEK
     | DAYOFYEAR | FROM_DAYS | FROM_UNIXTIME | HOUR | MAKEDATE | MAKETIME | MICROSECOND | MINUTE
     | MONTH | MONTHNAME | QUARTER | SECOND | SUBDATE | SYSDATE | TIME | TIME_TO_SEC | TIMESTAMP
-    | TO_DAYS | UNIX_TIMESTAMP | WEEK | YEAR
+    | TO_DAYS | UNIX_TIMESTAMP | UTC_DATE | UTC_TIME | UTC_DATETIME | WEEK | YEAR
```
Comment (acarbonetto) [inline sql/src/main/antlr/OpenSearchSQLParser.g4:400]: same here, why is it called UTC_DATETIME vs UTC_TIMESTAMP

**Review by acarbonetto — CHANGES_REQUESTED** [review]

File: docs/user/ppl/functions/datetime.rst (hunk 10 above)
Comment (MitchellGale-BitQuill) [reply docs/user/ppl/functions/datetime.rst:1210]: Good point, other way was following Java standard, not MySQL.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 11 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1037]: Timestamp is formatted as a datetime. We have no `exprTimeStamp`.

File: docs/user/dql/functions.rst (hunk 12 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst:1774]: Timestamp is formatted as a datetime. We have no `exprTimeStamp`.

File: core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java [hunk 15]
```diff
@@ -168,4 +168,10 @@ public class DateTimeFormatters {
           .appendPattern("uuuu-MM-dd HH:mm:ss[xxx]")
           .toFormatter()
           .withResolverStyle(ResolverStyle.STRICT);
+
+  // yyyy-MM-dd HH:mm:ss
+  public static final DateTimeFormatter DATE_TIME_FORMATTER_NON_STRUCT =
```
[the remark this answers is not in the document]
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java:173]: Addressed in 59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 4 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1007]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 5 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1018]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 6 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1028]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: docs/user/dql/functions.rst (hunk 7 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst:1721]: Made consistent with MySQL. Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: docs/user/dql/functions.rst (hunk 8 above)
Comment (MitchellGale-BitQuill) [reply docs/user/dql/functions.rst:1770]: Made consistent with MySQL. Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: docs/user/ppl/functions/datetime.rst (hunk 9 above)
Comment (MitchellGale-BitQuill) [reply docs/user/ppl/functions/datetime.rst:1161]: Made consistent with MySQL. Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: docs/user/ppl/functions/datetime.rst (hunk 10 above)
Comment (MitchellGale-BitQuill) [reply docs/user/ppl/functions/datetime.rst:1210]: Made consistent with MySQL. Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: ppl/src/main/antlr/OpenSearchPPLParser.g4 (hunk 13 above)
Comment (MitchellGale-BitQuill) [reply ppl/src/main/antlr/OpenSearchPPLParser.g4:401]: Good catch. Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: sql/src/main/antlr/OpenSearchSQLParser.g4 (hunk 14 above)
Comment (MitchellGale-BitQuill) [reply sql/src/main/antlr/OpenSearchSQLParser.g4:400]: Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/59ef32f37ea15e9f44cae06574c3cb9f27db26e1.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 16]
```diff
@@ -532,6 +536,30 @@ private FunctionResolver unix_timestamp() {
...
+    return define(BuiltinFunctionName.UTC_TIME.getName(),
+        impl(DateTimeFunction::exprUtcTime, TIME));
+  }
+
+  /**
+   * UTC_TIMESTAMP(). return the current UTC TimeStamp in format yyyy-MM-dd HH:mm:ss
+   */
+  private DefaultFunctionResolver utc_timestamp() {
+    return define(BuiltinFunctionName.UTC_TIMESTAMP.getName(),
+        impl(DateTimeFunction::exprUtcTimeStamp, TIMESTAMP));
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:560]: Return type should be `DATETIME`

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+  public void utcTime() {
+    FunctionExpression expr = dsl.utc_time();
+    LocalTime result = expr.valueOf(env).timeValue();
+    assertEquals(ExprCoreType.TIME, expr.type());
+
+    LocalDateTime currentUTC = LocalDateTime.now(ZoneOffset.UTC);
+    LocalTime lowerUTC = currentUTC.minus(allowedSecondsOffset, ChronoUnit.SECONDS).toLocalTime();
+    LocalTime upperUTC = currentUTC.minus(-allowedSecondsOffset, ChronoUnit.SECONDS).toLocalTime();
+
+    if (!lowerUTC.isBefore(result) || !upperUTC.isAfter(result)) {
```
Comment (forestmvey) [inline core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java:63]: Seems like a weird assertion, why not just something like `assertTrue(lowerUTC.isBefore(result) && upperUTC.isAfter(result));`

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java (hunk 2 above)
Comment (MitchellGale-BitQuill) [reply core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java]: Removed test UTCFunctionsTest.java to increase tests in with existing functions e5a95270b0c945d4541163359a480500f60e00ed

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java (hunk 17 above)
Comment (MitchellGale-BitQuill) [reply core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java]: UTCFunctionsTest.java was removed in e5a95270b0c945d4541163359a480500f60e00ed

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 16 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java]: Fixed in 2ea2397b1e79a33acd12f8468e235b9e5f125901

File: core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java (hunk 3 above)
Comment (MitchellGale-BitQuill) [reply core/src/test/java/org/opensearch/sql/expression/datetime/UTCFunctionsTest.java]: UTCFunctionsTest.java was removed in https://github.com/Bit-Quill/opensearch-project-sql/commit/e5a95270b0c945d4541163359a480500f60e00ed

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 1 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1008]: @Yury-Fridlyand any advice on how to address this if not including "yyyy-MM-dd"

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 11 above)
Comment (acarbonetto) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1034]: Would https://github.com/Bit-Quill/opensearch-project-sql/blob/bc91c66b572a564221ea87b7f126f2158e0ac083/core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java#L987 work?

**Review by acarbonetto — CHANGES_REQUESTED** [review]

**Comment by Yury-Fridlyand** [comment]
When you'll publish PR on upstream, please, mention that value of these functions is cached as it happens with `now`. Due to the limitations in the caching mechanism, the function can't accept the `fsp` argument.

File: docs/user/dql/functions.rst [hunk 18]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+    | utc_time()   |
+    |--------------|
+    | 17:54:27     |
+    +--------------+
+
+
+
+
+UTC_TIMESTAMP
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1765]:
```suggestion


UTC_TIMESTAMP
-------------
```

File: docs/user/ppl/functions/datetime.rst [hunk 19]
```diff
@@ -1152,6 +1152,81 @@ Example::
     +--------------------------+-----------------------------+
 
 
+UTC_DATE
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1156]:
```suggestion
UTC_DATE
--------
```

File: docs/user/ppl/functions/datetime.rst [hunk 20]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+    +--------------+
+    | UTC_DATE()   |
+    |--------------|
+    | 2022-10-03   |
+    +--------------+
+
+
+
+UTC_TIME
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1180]:
```suggestion
UTC_TIME
--------
```

File: docs/user/ppl/functions/datetime.rst [hunk 21]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+    | UTC_TIME()   |
+    |--------------|
+    | 17:54:27     |
+    +--------------+
+
+
+
+
+UTC_TIMESTAMP
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1205]:
```suggestion


UTC_TIMESTAMP
-------------
```

File: docs/user/ppl/functions/datetime.rst [hunk 22]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+    +---------------------+
+    | UTC_TIMESTAMP()     |
+    |---------------------|
+    | 2022-10-03 17:54:28 |
+    +---------------------+
+
+
+
+
+
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1229]:
```suggestion
```

File: docs/user/dql/functions.rst [hunk 23]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+    fetched rows / total rows = 1/1
+    +---------------------+
+    | utc_timestamp()     |
+    |---------------------|
+    | 2022-10-03 17:54:28 |
+    +---------------------+
+
+
+
+
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1788]:
```suggestion
```

File: docs/user/dql/functions.rst [hunk 24]
```diff
@@ -1711,6 +1711,81 @@ Example::
     +----------------------------+----------------------------+
 
 
+
+UTC_DATE
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1716]:
```suggestion
UTC_DATE
--------
```

File: core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java [hunk 25]
```diff
@@ -113,16 +142,18 @@ public void test_now_like_functions(Function<Expression[], FunctionExpression> f
...
-              extractValue(function.apply(new Expression[]{DSL.literal(0)})),
-              referenceGetter.get()
-      )) <= 1);
+          extractValue(function.apply(new Expression[]{DSL.literal(0)})),
+          referenceGetter.get()
+      )) <= delta);
     }
   }
-}
+}
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java:159]:
```suggestion
}

```

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 16 above)
Comment (Yury-Fridlyand) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:560]: Was it reverted?

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 26]
```diff
@@ -974,6 +1002,38 @@ private ExprValue exprTimeToSec(ExprValue time) {
...
+   * UTC_TIMESTAMP implementation for ExprValue.
+   *
+   * @return ExprValue.
+   */
+  private ExprValue exprUtcTimeStamp() {
+    String formattedLDT = LocalDateTime.now().format(DATE_TIME_FORMATTER_NON_STRICT);
+    ExprStringValue localDateValue = new ExprStringValue(formattedLDT);
+    ExprStringValue toTZ = new ExprStringValue("+00:00");
+
+    return exprTimestamp(exprDateTime(localDateValue, toTZ));
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1034]:
```suggestion
    return exprDateTime(localDateValue, toTZ);
```

File: docs/user/dql/functions.rst [hunk 27]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+    +--------------+
+    | utc_date()   |
+    |--------------|
+    | 2022-10-03   |
+    +--------------+
+
+
+
+UTC_TIME
+-------
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1740]:
```suggestion
UTC_TIME
--------
```

File: docs/user/dql/functions.rst [hunk 28]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+
+Returns the current UTC date as a value in 'YYYY-MM-DD'.
+
+Return type: DATE
+
+Specification: UTC_DATE() -> DATE
+
+Example::
+
+    >od SELECT UTC_DATE();
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1729]:
```suggestion
    > SELECT UTC_DATE();
```

File: docs/user/dql/functions.rst [hunk 29]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+
+Returns the current UTC time as a value in 'hh:mm:ss'.
+
+Return type: TIME
+
+Specification: UTC_TIME() -> TIME
+
+Example::
+
+    >od SELECT UTC_TIME();
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1753]:
```suggestion
    > SELECT UTC_TIME();
```

File: docs/user/dql/functions.rst [hunk 30]
```diff
@@ -1711,6 +1711,81 @@ Example::
...
+
+Returns the current UTC timestamp as a value in 'YYYY-MM-DD hh:mm:ss'.
+
+Return type: TIMESTAMP
+
+Specification: UTC_TIMESTAMP() -> TIMESTAMP
+
+Example::
+
+    >od SELECT UTC_TIMESTAMP();
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1778]:
```suggestion
    > SELECT UTC_TIMESTAMP();
```

File: docs/user/ppl/functions/datetime.rst [hunk 31]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+Returns the current UTC date as a value in 'YYYY-MM-DD'.
+
+Return type: DATE
+
+Specification: UTC_DATE() -> DATE
+
+Example::
+
+    >od source=people | eval `UTC_DATE()` = UTC_DATE() | fields `UTC_DATE()`
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1169]:
```suggestion
    > source=people | eval `UTC_DATE()` = UTC_DATE() | fields `UTC_DATE()`
```

File: docs/user/ppl/functions/datetime.rst [hunk 32]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+
+Returns the current UTC time as a value in 'hh:mm:ss'.
+
+Return type: TIME
+
+Specification: UTC_TIME() -> TIME
+
+Example::
+
+    >od source=people | eval `UTC_TIME()` = UTC_TIME() | fields `UTC_TIME()`
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1193]:
```suggestion
    > source=people | eval `UTC_TIME()` = UTC_TIME() | fields `UTC_TIME()`
```

File: docs/user/ppl/functions/datetime.rst [hunk 33]
```diff
@@ -1152,6 +1152,81 @@ Example::
...
+
+Returns the current UTC timestamp as a value in 'YYYY-MM-DD hh:mm:ss'.
+
+Return type: TIMESTAMP
+
+Specification: UTC_TIMESTAMP() -> TIMESTAMP
+
+Example::
+
+    >od source=people | eval `UTC_TIMESTAMP()` = UTC_TIMESTAMP() | fields `UTC_TIMESTAMP()`
```
Comment (Yury-Fridlyand) [inline docs/user/ppl/functions/datetime.rst:1218]:
```suggestion
    > source=people | eval `UTC_TIMESTAMP()` = UTC_TIMESTAMP() | fields `UTC_TIMESTAMP()`
```

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]

**Comment by Yury-Fridlyand** [comment]
Rebase and squash after all updates to ensure that you have DCO on the commit.

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 16 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java]: should be good now

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 26 above)
Comment (MitchellGale-BitQuill) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java]: Reverted change. Resolved.

File: core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java [hunk 34]
```diff
@@ -168,4 +168,10 @@ public class DateTimeFormatters {
           .appendPattern("uuuu-MM-dd HH:mm:ss[xxx]")
           .toFormatter()
           .withResolverStyle(ResolverStyle.STRICT);
+
+  // yyyy-MM-dd HH:mm:ss
+  public static final DateTimeFormatter DATE_TIME_FORMATTER_NON_STRICT =
+      new DateTimeFormatterBuilder()
+          .appendPattern("yyyy-MM-dd HH:mm:ss")
+          .toFormatter();
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/utils/DateTimeFormatters.java:176]: Please, make this one strict

File: core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java [hunk 35]
```diff
@@ -55,16 +57,43 @@ private static Stream<Arguments> functionNames() {
...
+  private static LocalTime getUtcTimeRefValue() {
+    return getUtcTimestampRefValue().toLocalTime();
   }
 
   private Temporal extractValue(FunctionExpression func) {
     switch ((ExprCoreType)func.type()) {
-      case DATE: return func.valueOf(null).dateValue();
-      case DATETIME: return func.valueOf(null).datetimeValue();
-      case TIME: return func.valueOf(null).timeValue();
+      case DATE:
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java:87]: You can revert this, no function returns `TIMESTAMP`.

File: common/src/main/java/org/opensearch/sql/common/grok/GrokCompiler.java [hunk 36]
```diff
@@ -0,0 +1,216 @@
...
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.sql.common.grok.exception.GrokException;
+
+public class GrokCompiler implements Serializable {
+
+  // We don't want \n and commented line
+  private static final Pattern patternLinePattern = Pattern.compile("^([A-z0-9_]+)\\s+(.*)$");
```
[the remark this answers is not in the document]
Comment (MaxKsyunz) [reply common/src/main/java/org/opensearch/sql/common/grok/GrokCompiler.java]: @MitchellGale-BitQuill I think this was fixed in upstream -- if you update the integ branch, the warnings will go away.

**Review by MaxKsyunz — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java [hunk 37]
```diff
@@ -113,16 +139,18 @@ public void test_now_like_functions(Function<Expression[], FunctionExpression> f
 
     // Check how calculations are precise:
     // `func()`
+    // 17:59:59.99999
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java:142]: Please, clean-up your notes or extend this to be a comment.

**Review by MitchellGale-BitQuill — COMMENTED** [review]
Addressed PR comments

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:1007]: You can mention return value type.

**Review by Yury-Fridlyand — APPROVED** [review]

File: core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java [hunk 38]
```diff
@@ -86,11 +107,11 @@ private long getDiff(Temporal sample, Temporal reference) {
   @ParameterizedTest(name = "{1}")
   @MethodSource("functionNames")
   public void test_now_like_functions(Function<Expression[], FunctionExpression> function,
-                       @SuppressWarnings("unused")  // Used in the test name above
-                       String name,
-                       ExprCoreType resType,
-                       Boolean hasFsp,
-                       Supplier<Temporal> referenceGetter) {
+                                      @SuppressWarnings("unused")  // Used in the test name above
```
Comment (forestmvey) [inline core/src/test/java/org/opensearch/sql/expression/datetime/NowLikeFunctionTest.java:110]: Was this indentation change needed?

**Review by forestmvey — APPROVED** [review]

**Pull request merged**